plugin/trino-hive-hadoop2/src/test/java/io/trino/plugin/hive/AbstractTestHiveFileSystemS3.java

```diff
@@ -93,4 +132,137 @@ public void testIgnoreHadoopFolderMarker()
+                Location.of(basePath.toString()),
+                trinoFileSystem,
+                new FileSystemDirectoryLister(),
+                new HdfsNamenodeStats(),
+                HiveFileIterator.NestedDirectoryPolicy.RECURSE);
+
+        List<String> recursiveListing = Streams.stream(recursiveIterator)
+                .map(TrinoFileStatus::getPath)
+                .toList();
+        // Should not include directories, or files underneath hidden directories
+        assertEqualsIgnoreOrder(
+                recursiveListing,
+                ImmutableList.of(
+                        format("%s/part=simple/plain-file.txt", basePath),
+                        format("%s/part=nested/parent/nested-file.txt", basePath),
+                        format("%s/part=plus+sign/plus-file.txt", basePath),
+                        format("%s/part=percent%%sign/percent-file.txt", basePath),
+                        format("%s/part=url%%20encoded/url-encoded-file.txt", basePath),
+                        format("%s/part=level1|level2/pipe-file.txt", basePath),
+                        format("%s/part=level1|level2/parent1/parent2/deeply-nested-file.txt", basePath),
+                        format("%s/part=level1 | level2/pipe-blanks-file.txt", basePath)));
+
+        HiveFileIterator shallowIterator = new HiveFileIterator(
+                fakeTable,
+                Location.of(basePath.toString()),
+                trinoFileSystem,
+                new FileSystemDirectoryLister(),
+                new HdfsNamenodeStats(),
+                HiveFileIterator.NestedDirectoryPolicy.IGNORED);
+        List<Path> shallowListing = Streams.stream(shallowIterator)
+                .map(TrinoFileStatus::getPath)
+                .map(Path::new)
+                .toList();
+        // Should not include any hidden files, folders, or nested files
+        assertEqualsIgnoreOrder(shallowListing, ImmutableList.of());
+    }
+
+    protected void createDirectory(String bucketName, String key)
+    {
+        // create meta-data for your folder and set content-length to 0
+        ObjectMetadata metadata = new ObjectMetadata();
+        metadata.setContentLength(0);
+        metadata.setContentType(DIRECTORY_MEDIA_TYPE.toString());
+        // create empty content
+        InputStream emptyContent = new ByteArrayInputStream(new byte[0]);
+        // create a PutObjectRequest passing the folder name suffixed by /
+        if (!key.endsWith("/")) {
+            key += "/";
+        }
+        PutObjectRequest putObjectRequest = new PutObjectRequest(bucketName, key, emptyContent, metadata);
+        // send request to S3 to create folder
+        s3Client.putObject(putObjectRequest);
+    }
+
+    protected void createFile(String bucketName, String key)
+    {
+        ObjectMetadata metadata = new ObjectMetadata();
+        metadata.setContentLength(0);
+        // create empty content
+        InputStream emptyContent = new ByteArrayInputStream(new byte[0]);
```

Comment:
Is it possible to use `nullInputStream` in InputStream class?

Reply:
Works.
Thanks for the suggestion.